src/main/java/org/terasology/las/bases/BaseProvider.java

```diff
@@ -43,12 +48,22 @@ public void setSeed(long seed) {
 
     @Override
     public void process(GeneratingRegion region) {
+        SurfacesFacet surfacesFacet = region.getRegionFacet(SurfacesFacet.class);
         Border3D border = region.getBorderForFacet(BaseFacet.class);
         BaseFacet facet = new BaseFacet(region.getRegion(), border);
 
         for (Base base : fixedBases) {
             facet.add(base);
         }
+
+        BlockRegion surfaceRegion = surfacesFacet.getWorldRegion();
+        for (Vector3ic pos : surfaceRegion) {
+            if ((redBaseRegion.contains(pos) || blackBaseRegion.contains(pos))) {
+                int y = surfacesFacet.getNextBelow(pos);
+                surfacesFacet.setWorld(pos.x(), y, pos.z(), false);
```

Comment:
I'm not sure what would happen if the `y` value is not part of the generating `region`. The implementation looks like it would not crash, but this might be a pitfall if the next surface layer is below the currently generated chunk...